src/app/coin/[coinId]/page.tsx

```diff
@@ -170,9 +169,57 @@ const DateText = styled.p`
   text-align: center;
 `;
 
-export default function Coin({ params }) {
+interface CoinProps {
+  params: { coinId: string };
+}
+
+interface CoinDataTypes {
+  id: string;
+  symbol: string;
+  name: string;
+  image: {
+    small: string;
+  };
+  links: {
+    homepage: string[];
+    blockchain_site: string[];
+  };
+  market_data: {
+    current_price: {
+      usd: number;
+    };
+    ath_date: {
+      usd: string;
+    };
+    atl_date: {
+      usd: string;
+    };
+    atl: {
+      usd: number;
+    };
+    ath: {
+      usd: number;
+    };
+    market_cap: {
+      usd: number;
+    };
+    fully_diluted_valuation: {
+      usd: number;
+    };
+    total_volume: {
+      usd: number;
```

Comment:
might wanna look into this when you will add more currencies: https://stackoverflow.com/questions/12710905/how-do-i-dynamically-assign-properties-to-an-object-in-typescript